website/addons/twofactor/static/twoFactorUserConfig.js

```diff
@@ -1,63 +1,204 @@
 'use strict';
 
+require('./user-cfg.css');
+
 var ko = require('knockout');
+var Raven = require('raven-js');
 var $ = require('jquery');
-require('./jquery.qrcode.min.js');
+var bootbox = require('bootbox');
+require('jquery-qrcode');
 
 var osfHelpers = require('js/osfHelpers');
 
-var SETTINGS_URL = '/api/v1/settings/twofactor/';
-
-function ViewModel(qrCodeSelector, otpURL) {
+function ViewModel(settingsUrl, qrCodeSelector) {
     var self = this;
+    self.settingsUrl = settingsUrl;
     self.qrCodeSelector = qrCodeSelector;
     self.tfaCode = ko.observable('');
 
     self.message = ko.observable('');
     self.messageClass = ko.observable('');
-    // Initialize QR Code
-    $(self.qrCodeSelector).qrcode(otpURL);
-
-    /** Change the flashed message. */
-    self.changeMessage = function(text, css, timeout) {
-        self.message(text);
-        var cssClass = css || 'text-info';
-        self.messageClass(cssClass);
-        if (timeout) {
-            // Reset message after timeout period
-            setTimeout(function() {
-                self.message('');
-                self.messageClass('text-info');
-            }, timeout);
+
+    self.isEnabled = ko.observable(false);
+    self.isConfirmed = ko.observable(false);
+    self.secret = ko.observable('');
+
+    self.urls = {};
+}
+
+ViewModel.prototype.initialize = function() {
+    var self = this;
+    return self.fetchFromServer().then(self.updateFromData.bind(self));
+};
+
+ViewModel.prototype.updateFromData = function(data) {
+    var self = this;
+    self.isEnabled(data.is_enabled);
+    self.isConfirmed(data.is_confirmed);
+    self.secret(data.secret);
+    self.urls = data.urls;
+    if (self.isEnabled()) {
+        // Initialize QR Code
+        if (!self.urls.otpauth) {
+            throw new Error('Tried to initailize jQuery.fn.qrcode without a otpauth URL');
+        } else {
+            $(self.qrCodeSelector).qrcode(self.urls.otpauth);
+        }
+    }
+};
+
+ViewModel.prototype.fetchFromServer = function() {
+    var self = this;
+    return $.getJSON(self.settingsUrl)
+        .then(function(response) {
+            return response.result;
+        })
+        .fail(function(xhr, status, error) {
+            Raven.captureMessage('Failed to fetch twofactor settings.', {
+                xhr: xhr,
+                status: status,
+                error: error
+            });
+            self.changeMessage('Could not retrieve Two-factor settings at ' +
+                'this time. Please refresh the page. ' +
+                'If the problem persists, email ' +
+                '<a href="mailto:support@osf.io">support@osf.io</a>.', 'text-danger', 5000);
+        });
+};
+
+/** Change the flashed message. */
+ViewModel.prototype.changeMessage = function(text, css, timeout) {
```

Comment:
Might be worth inheriting from `ChangeMessageMixin` to remove this duplicated code.